connectors/src/connectors/notion/temporal/workflows/index.ts

```diff
@@ -222,56 +266,100 @@ export async function notionGarbageCollectionWorkflow({
 
   const promises: Promise<void>[] = [];
 
-  // we go through each result page of the notion search API
-  do {
-    // It's a garbage collection, we want to fetch all pages.
-    const { pageIds, databaseIds, nextCursor } =
-      await getPagesAndDatabasesToSync({
-        connectorId,
-        lastSyncedAt: null,
-        // Only pass non-null cursors.
-        cursors: cursors.filter((c) => c !== null) as string[],
-        excludeUpToDatePages: false,
-        loggerArgs: {
+  async function runSearch(
```

Comment:
Any reason to not share this code with the one above?

Reply:
It was shared, but @Fraggle  removed the sharing because he found it confusing. So I'm not going to undo his recent change